src/test/java/hudson/plugins/git/extensions/impl/UserIdentityTest.java

```diff
@@ -1,9 +1,58 @@
 package hudson.plugins.git.extensions.impl;
 
+import hudson.EnvVars;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.TestGitRepo;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.GitSCMExtensionTest;
 import nl.jqno.equalsverifier.EqualsVerifier;
+import org.jenkinsci.plugins.gitclient.Git;
+import org.jenkinsci.plugins.gitclient.GitClient;
 import org.junit.Test;
 
-public class UserIdentityTest {
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class UserIdentityTest extends GitSCMExtensionTest  {
+
+    TestGitRepo repo;
+    GitClient git;
+
+    @Override
+    public void before() throws Exception {
+        repo = new TestGitRepo("repo", tmp.newFolder(), listener);
+        git = Git.with(listener, new EnvVars()).in(repo.gitDir).getClient();
+    }
+
+    @Override
+    protected GitSCMExtension getExtension() {
+        return new UserIdentity("Jane Doe", "janeDoe@xyz.com");
+    }
+
+    @Test
+    public void testUserIdentity() throws Exception {
+        FreeStyleProject projectWithMaster = setupBasicProject(repo);
+        UserIdentity userIdentity = new UserIdentity("Jane Doe", "janeDoe@xyz.com");
+        ((GitSCM)projectWithMaster.getScm()).getExtensions().add(userIdentity);
```

Comment:
The abstract class GitSCMExtensionTest already performs this add operation with setupBasicProject.  Is it intentional that it is called a second time?  If so, can you provide a comment that describes why it is called a second time?